src/operator/tensor/dot.cu

```diff
@@ -23,10 +23,142 @@
  */
 
 #include "./dot-inl.h"
+#include <cub/cub.cuh>
 
 namespace mxnet {
 namespace op {
 
+template<typename gpu>
```

Comment:
It looks like for dot, gpu kernels are in `dot-inl.cuh`. Let's move the implementation here